test/integrationTests/commandEnablement.integration.test.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect, test, beforeAll, afterAll, describe } from '@jest/globals';
+import * as vscode from 'vscode';
+import { activateCSharpExtension } from './integrationHelpers';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+
+describe(`Command Enablement: ${testAssetWorkspace.description}`, function () {
+    beforeAll(async function () {
+        await activateCSharpExtension();
+    });
+
+    afterAll(async () => {
+        await testAssetWorkspace.cleanupWorkspace();
+    });
+
+    test('Only expected commands are available', async function () {
+        const commands = await vscode.commands.getCommands(true);
+
+        // Ensure the standalone Roslyn commands are available.
+        expect(commands).toContain('dotnet.openSolution');
+        expect(commands).toContain('dotnet.restartServer');
+        expect(commands).toContain('dotnet.generateAssets');
+        expect(commands).toContain('dotnet.restore.project');
+        expect(commands).toContain('dotnet.restore.all');
+        expect(commands).toContain('dotnet.test.runTestsInContext');
+        expect(commands).toContain('dotnet.test.debugTestsInContext');
+        expect(commands).toContain('csharp.listProcess');
+        expect(commands).toContain('csharp.listRemoteProcess');
+        expect(commands).toContain('csharp.listRemoteDockerProcess');
+        expect(commands).toContain('csharp.attachToProcess');
+        expect(commands).toContain('csharp.reportIssue');
```

Comment:
Since we have a very similar list in the omnisharp side of the tests, is it worth just having a shared file that list the commands that are:

1. Expected to be in both.
2. Expected to be in Roslyn only
3. Expected to be in Omnisharp only

That way if nothing else there's not a risk of updating one list but forgetting to update the other.